macros.d/macros.systemd

```diff
@@ -0,0 +1,176 @@
+#  -*- Mode: rpm-spec; indent-tabs-mode: nil -*- */
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#  Copied from https://github.com/systemd/systemd/blob/main/src/rpm/macros.systemd.in
+
+# RPM macros for packages installing systemd unit files
+
+%_systemd_util_dir /usr/lib/systemd
+%_unitdir /usr/lib/systemd/system
+%_userunitdir /usr/lib/systemd/user
+%_presetdir /usr/lib/systemd/system-preset
+%_userpresetdir /usr/lib/systemd/user-preset
+%_udevhwdbdir /usr/lib/udev/hwdb.d
+%_udevrulesdir /usr/lib/udev/rules.d
+%_journalcatalogdir /usr/lib/systemd/catalog
+%_binfmtdir /usr/lib/binfmt.d
+%_sysctldir /usr/lib/sysctl.d
+%_sysusersdir /usr/lib/sysusers.d
+%_tmpfilesdir /usr/lib/tmpfiles.d
+%_environmentdir /usr/lib/environment.d
+%_modulesloaddir /usr/lib/modules-load.d
+%_modprobedir /usr/lib/modprobe.d
+%_systemdgeneratordir /usr/lib/systemd/system-generators
+%_systemdusergeneratordir /usr/lib/systemd/user-generators
+%_systemd_system_env_generator_dir /usr/lib/systemd/system-environment-generators
+%_systemd_user_env_generator_dir /usr/lib/systemd/user-environment-generators
+
+# Because we had one release with a typo...
+# This is temporary (Remove after systemd 240 is released)
+%_environmnentdir %{warn:Use %%_environmentdir instead}%_environmentdir
+
+%systemd_requires \
+Requires(post): systemd \
+Requires(preun): systemd \
+Requires(postun): systemd \
+%{nil}
+
+%systemd_ordering \
+OrderWithRequires(post): systemd \
+OrderWithRequires(preun): systemd \
+OrderWithRequires(postun): systemd \
+%{nil}
+
+%__systemd_someargs_0(:) %{error:The %%%1 macro requires some arguments}
+%__systemd_twoargs_2() %{nil}
+
+%systemd_post() \
+%{expand:%%{?__systemd_someargs_%#:%%__systemd_someargs_%# systemd_post}} \
+if [ $1 -eq 1 ] && [ -x "/usr/lib/systemd/systemd-update-helper" ]; then \
+    # Initial installation \
+    /usr/lib/systemd/systemd-update-helper install-system-units %{?*} || : \
```

Comment:
This update helper does not exist in older distros such as EL7. That will mean the actual code isn't executed. It may be safer to take the least common denominator, like EL7 which has a fairly old systemd (by now).

Reply:
EL7 is no longer supported by Red Hat as of August 30, 2021, right? Thus the Jenkins project does not support it, either.

Reply:
That's not really correct. Per https://access.redhat.com/product-life-cycles/ EL7 is supported until 2024.

That said, the update helper is quite new. It was introduced in systemd 250 (https://github.com/systemd/systemd/commit/6d825ab2d42d3219e49a192bf99f9c09134a0df4) and EL8 has systemd 239. Even Fedora 35 (the latest stable) only has systemd 249.

Reply:
Centos 7 still has 'support' till the rhel date too https://endoflife.date/centos

I suspect there's still quite a lot of centos 7 distributions out there because of the centos 8 fiasco.

(and we still ship centos 7 in the docker images)

Reply:
> (and we still ship centos 7 in the docker images)

This is unrelated to the RPM macros: centos 7 image is not built for systemd and should not be used for this unless edge cases (such as simulating VMs).

Reply:
somewhat related in a support discussion though. Of course systemd isn't used there for managing the process 😄